controller_manager/test/mode_switch_test.cpp

```diff
@@ -0,0 +1,380 @@
+#include <gtest/gtest.h>
+
+#include <map>
+#include <set>
+#include <string>
+#include <algorithm>
+#include <iterator>
+
+#include <controller_manager/controller_manager.h>
+#include <hardware_interface/robot_hw.h>
+#include <hardware_interface/joint_command_interface.h>
+
+namespace hardware_interface {
+    bool operator<(hardware_interface::ControllerInfo const& i1, hardware_interface::ControllerInfo const& i2){
+        return i1.name < i2.name;
+    }
+}
+template<typename T> T intersect(const T& v1, const T &v2){
```

Comment:
Style: In `ros_control` opening braces go in new lines, except in one-liners (there are a few violations out there that have slipped).